src/bytes.rs

```diff
@@ -507,6 +508,46 @@ impl Bytes {
         self.truncate(0);
     }
 
+    /// Try to convert self into `BytesMut`.
+    ///
+    /// If `self` is unique, this will succeed and return a `BytesMut` with
+    /// the contents of `self` without copying. If `self` is not unique, this
+    /// will fail and return self.
```

Comment:
I think it would be nice to elaborate in the docs that it must be unique for the entire allocation, not just this particular range.

(for both methods)